packages/neon_framework/lib/src/models/account.dart

```diff
@@ -114,6 +114,18 @@ class Account implements Credentials, Findable {
     return result;
   }
 
+  /// Get the necessary `Authorization` headers for a given [uri].
+  ///
+  /// This method ensures no credentials are sent to the wrong server.
+  Map<String, String> getAuthorizationHeaders(Uri uri) {
+    // Only send the authentication headers when sending the request to the server of the account
+    if (uri.toString().startsWith(serverURL.toString()) && client.authentications.isNotEmpty) {
+      return client.authentications.first.headers;
+    }
+
+    return <String, String>{};
+  }
+
```

Comment:
```suggestion
  Map<String, String>? getAuthorizationHeaders(Uri uri) {
    // Only send the authentication headers when sending the request to the server of the account
    if (uri.toString().startsWith(serverURL.toString())) {
      return client.authentications.firstOrNull?.headers;
    }

    return null;
  }
```

I would prefer null. Feel free to make the parameter for the dynamite client nullable (I wanted to change for the non raw method anyways).

Reply:
Agreed, I only didn't do it because the request method expected a non-null value (which you fixed now).